src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CommitManagerProvider.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.Language.Intellisense.AsyncCompletion;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.Completion.AsyncCompletion
+{
+    [Export(typeof(IAsyncCompletionCommitManagerProvider))]
+    [Name("Roslyn Completion Commit Manager")]
+    [ContentType(ContentTypeNames.RoslynContentType)]
+    internal class CommitManagerProvider : IAsyncCompletionCommitManagerProvider
+    {
+        private readonly IAsyncCompletionCommitManager _instance = new CommitManager();
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CommitManagerProvider()
```

Comment:
@ivanbasov For the thread expectation validation concerns, try adding an `IThreadingContext threadingContext` parameter here (which will hopefully get imported without issue) to then pass to the actual `CommitManager` which would then be able to inherit from `ForegroundThreadAffinitizedObject` and do the proper checks. And apply this pattern to the other exported providers in this directory where needed.

Reply:
Thank you, @dpoeschl!

---
In reply to: [227947931](https://github.com/dotnet/roslyn/pull/29016#discussion_r227947931) [](ancestors = 227947931)